File: common/src/main/java/com/ssblur/scriptor/word/descriptor/power/SimpleStrengthDescriptor.java
package com.ssblur.scriptor.word.descriptor.power;

import com.ssblur.scriptor.word.descriptor.Descriptor;

public class SimpleStrengthDescriptor extends Descriptor implements StrengthDescriptor {
  Cost cost;
  double strength;
  boolean allowDuplication = false;

  public SimpleStrengthDescriptor(int cost, double strength) {
    this.cost = new Cost(cost, COSTTYPE.ADDITIVE);
    this.strength = strength;
  }

  @Override
  public Cost cost() {
    return cost;
  }

  @Override
  public double strengthModifier() {
    return strength;
  }

  public SimpleStrengthDescriptor allowDuplication() {
    allowDuplication = true;
    return this;
  }

  @Override
  public boolean allowsDuplicates() {
    return allowDuplication;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/power/SimpleStrengthDescriptor.kt
package com.ssblur.scriptor.word.descriptor.power

import com.ssblur.scriptor.api.word.Descriptor

class SimpleStrengthDescriptor(cost: Int, var strength: Double) : Descriptor(), StrengthDescriptor {
    var cost: Cost = Cost(cost.toDouble(), COSTTYPE.ADDITIVE)
    var allowDuplication: Boolean = false

    override fun cost(): Cost {
        return cost
    }

    override fun strengthModifier(): Double {
        return strength
    }

    fun allowDuplication(): SimpleStrengthDescriptor {
        allowDuplication = true
        return this
    }

    override fun allowsDuplicates(): Boolean {
        return allowDuplication
    }
}
